File: 724. Find Pivot Index.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        aggr = sum(nums)
        prefix = 0
        for i in range(len(nums)):
            if prefix == aggr - prefix - nums[i]:
                return i
            prefix += nums[i]
        return -1

File: 724-find-pivot-index/724-find-pivot-index.cpp
class Solution {
public:
    int pivotIndex(vector<int>& nums) {
        int n = nums.size();
        vector<int>prefix(n+1);
        
        for (int i = 1; i <= n; i++) {
            prefix[i] = prefix[i-1] + nums[i-1];
        }
        
        for (int i = 1; i <= n; i++) {
            if (prefix[i-1] == prefix[n] - prefix[i])
                return i-1;
        }
        return -1;
    }
};